metadata/metadata_test.go

```diff
@@ -555,6 +555,126 @@ var _ = Describe("Metadata", func() {
 		})
 	})
 
+	When("SetAnnotation is called", func() {
+
+		It("should error if the object is nil", func() {
+			err := SetAnnotation(nil, "foo", "bar")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("object cannot be nil"))
+		})
+
+		It("should add the annotation to the object", func() {
+			annotations := map[string]string{"foo": "bar"}
+			pod := &corev1.Pod{}
+
+			Expect(pod.Annotations).To(HaveLen(0))
+			for key, value := range annotations {
+				Expect(SetAnnotation(pod, key, value)).To(Succeed())
+			}
+			for key, value := range annotations {
+				Expect(pod.Annotations).To(HaveKeyWithValue(key, value))
+			}
+		})
+
+		It("should add the annotation to the existing ones", func() {
+			annotations := map[string]string{"foo": "bar"}
+			pod := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"quux": "corge"},
+				},
+			}
+
+			Expect(pod.Annotations).To(HaveLen(1))
+			for key, value := range annotations {
+				Expect(SetAnnotation(pod, key, value)).To(Succeed())
+			}
+			Expect(pod.Annotations).To(HaveLen(2))
+			Expect(pod.Annotations).To(gstruct.MatchAllKeys(gstruct.Keys{
+				"foo":  Equal("bar"),
+				"quux": Equal("corge"),
+			}))
+		})
+
+		It("should update the existing annotation", func() {
+			annotations := map[string]string{"foo": "bar"}
+			pod := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"foo": "corge"},
+				},
+			}
+
+			Expect(pod.Annotations).To(HaveLen(1))
+			for key, value := range annotations {
+				Expect(SetAnnotation(pod, key, value)).To(Succeed())
+			}
+			Expect(pod.Annotations).To(HaveLen(1))
+			for key, value := range annotations {
+				Expect(pod.Annotations).To(HaveKeyWithValue(key, value))
+			}
```

Comment:
If it has a single element at this point you could just check for the specific element without the loop:
```
Expect(pod.Annotations["foo"]).To(Equal("corge"))
```

Reply:
I was just trying to remove the second loop, but fair point.